spikeinterface/comparison/groundtruthstudy.py

```diff
@@ -203,28 +199,39 @@ def aggregate_dataframes(self, copy_into_folder=True, **karg_thresh):
 
         return dataframes
 
-    def compute_metrics(self, rec_name, metric_names=['snr'],
+    def compute_metrics(self, rec_name, sorter_name=None, metric_names=['snr'],
                 ms_before=3., ms_after=4., max_spikes_per_unit=500, 
-                n_jobs=-1, total_memory='1G', **snr_kargs):
+                n_jobs=-1, total_memory='1G', templates_only=False, **snr_kargs):
 
         rec = self.get_recording(rec_name)
-        gt_sorting = self.get_ground_truth(rec_name)
-        
+        if sorter_name is None:
+            sorting = self.get_ground_truth(rec_name)
+            sorter_name = 'gt'
+        else:
+            assert sorter_name in self.sorter_names
+            sorting = self.get_sorting(sorter_name, rec_name)
+
         # waveform extractor
-        waveform_folder = self.study_folder / 'metrics' / f'waveforms_{rec_name}'
+        waveform_folder = self.study_folder / 'metrics' / f'waveforms_{sorter_name}_{rec_name}'
+
         if waveform_folder.is_dir():
             shutil.rmtree(waveform_folder)
-        we = WaveformExtractor.create(rec, gt_sorting, waveform_folder)
+        we = WaveformExtractor.create(rec, sorting, waveform_folder)
         we.set_params(ms_before=ms_before, ms_after=ms_after, max_spikes_per_unit=max_spikes_per_unit)
         we.run(n_jobs=n_jobs, total_memory=total_memory)
         
         # metrics
-        metrics = compute_quality_metrics(we, metric_names=metric_names)
-        filename = self.study_folder / 'metrics' / f'metrics _{rec_name}.txt'
-        metrics.to_csv(filename, sep='\t', index=True)
+        if templates_only:
```

Comment:
don't make sens with this method name.
Maybe you can create 2 methods one for waveform and one for metrics.